queue_job/jobrunner/channels.py

```diff
@@ -980,16 +982,36 @@ def get_channel_by_name(self, channel_name, autocreate=False):
         >>> c = cm.get_channel_by_name('sub')
         >>> c.fullname
         'root.sub'
+        >>> c = cm.get_channel_by_name('root.sub.not.configured', parent_fallback=True)
+        >>> c.fullname
+        'root.sub.sub.not.configured'
         """
         if not channel_name or channel_name == self._root_channel.name:
             return self._root_channel
         if not channel_name.startswith(self._root_channel.name + "."):
             channel_name = self._root_channel.name + "." + channel_name
         if channel_name in self._channels_by_name:
             return self._channels_by_name[channel_name]
-        if not autocreate:
+        if not autocreate and not parent_fallback:
             raise ChannelNotFound("Channel %s not found" % channel_name)
         parent = self._root_channel
+        if parent_fallback:
+            # Look for first direct parent w/ config.
+            # Eg: `root.edi.foo.baz` will falback on `root.edi.foo`
+            # or `root.edi` or `root` in sequence
+            parent_name = channel_name
+            while True:
+                parent_name = parent_name.rsplit(".", 1)[:-1][0]
```

Comment:
And if `channel_name == "root"`?
```python
>>> parent_name = "root"
>>> parent_name.rsplit(".", 1)[:-1][0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: list index out of range
```

Reply:
Ah there is a safeguard on the first line to avoid it... nevermind!

Reply:
yes :smile: